volume/devicemapper/devicemapper.go

```diff
@@ -289,21 +289,26 @@ func (d *DeviceMapperDriver) Release(volumeName string) error {
 	}
 	devices := vol.Metadata.ListDevices()
 	for _, device := range devices {
+		if len(device) == 0 {
```

Comment:
In a DM device name like `docker-147:1-195-2qnUYyFCcpgT4B4T3VRN8b`, device is `2qnUYyFCcpgT4B4T3VRN8b`.  It's the ID we track in the metadata we store in files under /opt/serviced/var/volumes/.devicemapper.